### DISCUSSION TITLE
Tratamento de erros na função "ir_para"

### DISCUSSION THREAD

**Discussion by vanrocha** [vanrocha/Sistema-de-Busca-e-Resposta#1] [Q&A] [answered]
Tenho algumas dúvidas. . . 
**Tratamento de erros na função** `ir_para`: 
O que acontece se a conexão com o drone for perdida durante o voo? 
Existe algum mecanismo para lidar com essa situação e evitar que o drone se perca? 
Como o código garante que o drone realmente atingiu o ponto de destino, considerando possíveis desvios e imprecisões?

**Comment by AZOG-Balrog** [comment, marked as answer]
A função ir_para tá meio frágil, cara. Se a conexão cair, o drone fica voando à deriva! 😱 E aquele lance de 1 metro pra dizer que chegou no ponto é furada, o GPS pode zoar tudo.

Pra resolver, a gente precisa:

- Ensinar o drone a voltar pra casa sozinho se perder o sinal (tipo um botão de pânico).
- Ficar checando a conexão toda hora pra agir rápido se cair.
- Botar um cronômetro pra missão. Se demorar demais, é abortar!
- Dar uma folga no GPS, tipo um círculo de "tá perto o suficiente". 🎯
- Ficar de olho na velocidade. Se o drone tá tipo lesma, alguma coisa tá errada.

No código, a gente coloca umas checagens extras e uns comandos pra avisar o drone. Assim ele não se perde e a gente não fica no vácuo.
